daemon/image_delete.go

```diff
@@ -207,7 +207,7 @@ func (daemon *Daemon) imageDeleteHelper(img *image.Image, records *[]types.Image
 	// First, determine if this image has any conflicts. Ignore soft conflicts
 	// if force is true.
 	if conflict := daemon.checkImageDeleteConflict(img, force); conflict != nil {
-		if quiet && !daemon.imageIsDangling(img) {
+		if quiet && (!daemon.imageIsDangling(img) || strings.Contains(conflict.message, "being used")) {
```

Comment:
We should not rely on string comparisons here. This will break if the error message changes.

Instead, let's add a field to `imageDeleteConflict` and use that.